pkg/karpenter/karpenter.go

```diff
@@ -64,17 +64,14 @@ func (k *Installer) Install(ctx context.Context, serviceAccountRoleARN string, i
 	}
 	serviceAccountMap := map[string]interface{}{
 		create: api.IsEnabled(k.ClusterConfig.Karpenter.CreateServiceAccount),
-	}
-	if serviceAccountRoleARN != "" {
-		serviceAccountMap[serviceAccountAnnotation] = map[string]interface{}{
+		serviceAccountAnnotation: map[string]interface{}{
 			api.AnnotationEKSRoleARN: serviceAccountRoleARN,
-		}
+		},
+		serviceAccountName: DefaultServiceAccountName,
 	}
 	values := map[string]interface{}{
-		controller: map[string]interface{}{
-			clusterName:     k.ClusterConfig.Metadata.Name,
-			clusterEndpoint: k.ClusterConfig.Status.Endpoint,
-		},
+		clusterName:     k.ClusterConfig.Metadata.Name,
+		clusterEndpoint: k.ClusterConfig.Status.Endpoint,
```

Comment:
Is there validation for `Status` so that this doesn't panic?

Reply:
I think so, yeah, but I forgot where. :D

Reply:
Ah yes, it's in refresh cluster status.